app/utils/transformers/HouseConsignmentAdditionalReferenceTransformer.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils.transformers
+
+import connectors.ReferenceDataConnector
+import generated.AdditionalReferenceType02
+import models.reference.AdditionalReferenceType
+import models.{Index, UserAnswers}
+import pages.houseConsignment.index.items.additionalReference.AdditionalReferencePage
+import uk.gov.hmrc.http.HeaderCarrier
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class HouseConsignmentAdditionalReferenceTransformer @Inject() (
+  referenceDataConnector: ReferenceDataConnector
+)(implicit ec: ExecutionContext)
+    extends PageTransformer {
+
+  private case class TempAdditionalReference(typeValue: AdditionalReferenceType)
+
+  def transform(additionalReferences: Seq[AdditionalReferenceType02], hcIndex: Index, itemIndex: Index)(implicit
```

Comment:
This method could be moved to the pre-existing `AdditionalReferenceTransformer`. No real need to have both :)